kyo-core/shared/src/main/scala/kyo/ios.scala

```diff
@@ -36,22 +36,15 @@ sealed trait IOs extends Effect[IOs]:
         }
 
     def attempt[T, S](v: => T < S): Try[T] < S =
-        eval(v.map(Success(_): Try[T]))((s, k) =>
-            try k()
-            catch
-                case ex if NonFatal(ex) =>
-                    Failure(ex)
-        )
+        core.catching(v.map(Success(_): Try[T])) {
+            case ex if NonFatal(ex) =>
+                Failure(ex)
+        }
```

Comment:
There is no longer a need to match on `NonFatal` here - that's already checked in the loop.